Makefile.toml

```diff
@@ -232,1396 +242,691 @@ fi
+command = "${TWOLITER}"
+args = [
+    "--log-level=${TWOLITER_LOG_LEVEL}",
+    "exec",
+    "setup",
+    "--project-path=${TWOLITER_PROJECT}",
+    "${@}",
+]
+
 [tasks.setup-build]
-dependencies = ["setup"]
-script = [
-'''
-for cmd in docker gzip lz4; do
-  if ! command -v ${cmd} >/dev/null 2>&1 ; then
-    echo "required program '${cmd}' not found" >&2
-    exit 1
-  fi
-done
-'''
+dependencies = ["install-twoliter"]
+command = "${TWOLITER}"
+args = [
+    "--log-level=${TWOLITER_LOG_LEVEL}",
+    "exec",
+    "setup-build",
+    "--project-path=${TWOLITER_PROJECT}",
+    "--cargo-home=${CARGO_HOME}",
+    "${@}",
 ]
 
 [tasks.fetch]
-dependencies = [
-  "fetch-sdk",
-  "fetch-toolchain",
-  "fetch-sources",
-  "fetch-vendored",
+dependencies = ["install-twoliter"]
+command = "${TWOLITER}"
+args = [
+    "--log-level=${TWOLITER_LOG_LEVEL}",
+    "exec",
+    "fetch",
+    "--project-path=${TWOLITER_PROJECT}",
+    "--cargo-home=${CARGO_HOME}",
+    "${@}",
 ]
 
 [tasks.fetch-sdk]
-dependencies = ["setup-build"]
-script_runner = "bash"
-script = [
-'''
-if ! docker image inspect "${BUILDSYS_SDK_IMAGE}" >/dev/null 2>&1 ; then
-  if ! docker pull "${BUILDSYS_SDK_IMAGE}" ; then
-    echo "failed to pull '${BUILDSYS_SDK_IMAGE}'" >&2
-    exit 1
-  fi
-fi
-'''
+dependencies = ["install-twoliter"]
+command = "${TWOLITER}"
+args = [
+    "--log-level=${TWOLITER_LOG_LEVEL}",
+    "exec",
+    "fetch-sdk",
+    "--project-path=${TWOLITER_PROJECT}",
+    "--cargo-home=${CARGO_HOME}",
+    "${@}",
 ]
 
 [tasks.fetch-toolchain]
-dependencies = ["setup-build"]
-script_runner = "bash"
-script = [
-'''
-if docker image inspect "${BUILDSYS_TOOLCHAIN}-${BUILDSYS_ARCH}" >/dev/null 2>&1 ; then
-  exit 0
-fi
-
-case "${BUILDSYS_ARCH}" in
-  x86_64) docker_arch="amd64" ;;
-  aarch64) docker_arch="arm64" ;;
-esac
-
-# We want the image with the target's native toolchain, rather than one that matches the
-# host architecture.
-if ! docker pull --platform "${docker_arch}" "${BUILDSYS_TOOLCHAIN}" ; then
-  echo "could not pull '${BUILDSYS_TOOLCHAIN}' for ${docker_arch}" >&2
-  exit 1
-fi
-
-# Apply a tag to distinguish the image from other architectures.
-if ! docker tag "${BUILDSYS_TOOLCHAIN}" "${BUILDSYS_TOOLCHAIN}-${BUILDSYS_ARCH}" ; then
-  echo "could not tag '${BUILDSYS_TOOLCHAIN}-${BUILDSYS_ARCH}'" >&2
-  exit 1
-fi
-'''
+dependencies = ["install-twoliter"]
+command = "${TWOLITER}"
+args = [
+    "--log-level=${TWOLITER_LOG_LEVEL}",
+    "exec",
+    "fetch-toolchain",
+    "--project-path=${TWOLITER_PROJECT}",
+    "--cargo-home=${CARGO_HOME}",
+    "${@}",
 ]
 
 [tasks.fetch-sources]
-dependencies = ["setup"]
-script_runner = "bash"
-script = [
-'''
-for ws in sources variants tools; do
-  [ -d "${ws}" ] || continue
-  cargo fetch --locked --manifest-path ${ws}/Cargo.toml
-done
-chmod -R o+r ${CARGO_HOME}
-'''
+dependencies = ["install-twoliter"]
+command = "${TWOLITER}"
+args = [
+    "--log-level=${TWOLITER_LOG_LEVEL}",
+    "exec",
+    "fetch-sources",
```

Comment:
This `fetch-sources` token should be available as `CARGO_MAKE_CURRENT_TASK_NAME`.

I wonder if it would be possible to streamline these changes with a generic task, along the lines of what `tasks.publish-setup-without-key` does - by setting a variable and then running a different task.

That could potentially clean up a lot of the repetition in this change.